packages/three-vrm/src/VRMUtils/index.ts

```diff
@@ -3,6 +3,8 @@ import { removeUnnecessaryJoints } from './removeUnnecessaryJoints';
 import { removeUnnecessaryVertices } from './removeUnnecessaryVertices';
 import { rotateVRM0 } from './rotateVRM0';
 
+export type { deepDispose, removeUnnecessaryJoints, removeUnnecessaryVertices, rotateVRM0 };
```

Comment:
why is the line needed?

Reply:
see https://github.com/pixiv/three-vrm/pull/1293#issuecomment-1740419724

Reply:
mhm,,, How does it work? If I understand that the api-extractor simply forgets to output types for such static members, is that correct?

Reply:
I added these to avoid api-extractor generating dts with some missing types. But after trying again now the issue has not been fixed. 🤔 Let me check it again.

Reply:
Ah, I think I finally figured this out.

1. `three-vrm` is exporting `VRMExpressionLoaderPlugin` form `three-vrm-core` which use `types-vrm-0.0` inside 
2. if we bundle `three-vrm-core` with `types-vrm-0.0` first, some exports from `types-vrm-0.0` becames private in `three-vrm-core` but it works fine
4. if we bundle `three-vrm` with bundled `three-vrm-core` then, the private part is untouchable from `three-vrm` so the output dts is incomplete
    1. which seems reasonable so it's hard to say it's a bug in `api-extractor`

The same thing happends to `ConstraintSchema`.

So our options are

1. do not bundle dts from `types-*` packages in intermediate packages `three-vrm-*` and only bundle them inside  `three-vrm`
    1. can achieve single dts and no deps install
    2. some exports from `types-*` are still private so maybe hard to use
    3. types may conflict to user installed `types-*`
    4. introduce inconstancy
2. treat all `types-*` as dependencies or peerdependencies and let user install them
    1. not true dts bundle but flexible
    2. option to not install `types-vrm-0.0` at all
5. re-export or provide a public interface for each `types-*` in `three-vrm-*`
    1. can achieve single dts and no deps install
    3. needs dependency design?

What do you think?

Reply:
Sounds like 2 is the best option...? It sounds easier from a contributor perspective at least.
3 is probably the most favorable option from a user perspective but it would make the module hard to maintain.